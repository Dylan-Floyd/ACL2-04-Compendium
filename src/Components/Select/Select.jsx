import React from 'react'

//This component can accept either an optionsList or optionsObject
//optionsList should be an array of strings,
//    if optionsList is provided, the option elements will have their value and label set to the string
//optionsObject should be an object with the properties representing the value and label,
//    if optionsObject is provided, the option elements will have their label set to the property key, and the value set to the property value
//    as in the object should be formatted like this:
//    optionsObject = {
//        MyOptionLabel: myOptionValue,
//        OptionTwoLabel: option2Value
//    }
export default function Select({
  label,
  selectedOption,
  optionsList,
  optionsObject,
  handleChange
}) {
  let optionEls = []
  if (Array.isArray(optionsList)) {
    optionEls = optionsList.map(option => <option value={option} key={option}>{option}</option>)
  } else if (optionsObject !== undefined) {
    optionEls = Object.entries(optionsObject).map(([uiLabel, apiValue]) => <option value={apiValue} key={apiValue}>{uiLabel}</option>)
  }
  
  return (
    <label>
      {label}
      <select value={selectedOption} onChange={e => handleChange(e.target.value)}>
        {optionEls}
      </select>
    </label>
  )
}
