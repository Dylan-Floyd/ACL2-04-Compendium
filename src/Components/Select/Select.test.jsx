import { render } from '@testing-library/react'
import Select from './Select.jsx'

it('Should render a Select component', () => {
  const { container } = render(<Select
    label='bob'
    selectedOption='1'
    optionsList={['1', '2']}
  />)
  expect(container).toMatchSnapshot()
})