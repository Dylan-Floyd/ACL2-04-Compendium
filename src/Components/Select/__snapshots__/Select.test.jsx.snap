// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should render a Select component 1`] = `
<div>
  <label>
    bob
    <select>
      <option
        value="1"
      >
        1
      </option>
      <option
        value="2"
      >
        2
      </option>
    </select>
  </label>
</div>
`;
